force-app/main/default/classes/UpdateContactPhoneNumber.cls

```diff
@@ -17,15 +17,23 @@ public class UpdateContactPhoneNumber implements Database.Batchable<sObject>, Da
 
     public void execute(Database.BatchableContext bc, List<Contact> scope){
 
-        List<Contact> contactsToBeUpdated = new List<Contact>();
-
-        for(Contact contactRecord : scope){
-            contactRecord.Phone = phoneToBeUpdated;
-            recordsProcessed++;
-            contactsToBeUpdated.Add(contactRecord);
-        }    
-
-        update contactsToBeUpdated;
+        try{
+            List<Contact> contactsToBeUpdated = new List<Contact>();
+
+            for(Contact contactRecord : scope){
+                contactRecord.Phone = phoneToBeUpdated;
+                recordsProcessed++;
+                contactsToBeUpdated.Add(contactRecord);
+            }    
+    
+            update contactsToBeUpdated;
+        }
+        catch(Exception ex){
+			Error_Log__c errorLog = new Error_Log__c();
+            errorLog.Name = 'UpdateContactPhoneNumber';
+            errorLog.StackError__c = ex.getStackTraceString();   
+            errorLog.Message__c = ex.getMessage();
```

Comment:
With this approach for each catch block, we would have to do the same steps, duplicating the code.
Please create a new class HandleException which will contain a method logException which will receive as parameter an Exception e . Move this logic there and make sure that the errorLog is inserted.